src/firmware.c

```diff
@@ -38,7 +38,15 @@ int main(void)
     screen_splash();
     qtouch_init();
     common_main();
-    bitbox02_smarteeprom_init();
+    if (!bitbox02_smarteeprom_init()) {
+        /*
+         * Incorrect version!
+         * Something has gone terribly wrong.
+         * Maybe reset the whole device?
+         */
+        common_main_bootloader_autoenter();
+        Abort("Unrecognized SmartEEPROM version.");
```

Comment:
would it be useful for support to show the current version in the message? `Unrecognized version: 0x123.`

Reply:
I actually had to go the AbortAutoenter route to make this happen, which seems cleaner anyway. Please check the new version here:

https://github.com/digitalbitbox/bitbox02-firmware/pull/816